### PR TITLE
new package: gitoxide

### PR DESCRIPTION
**Pull request by SunPodder**

### CODE REVIEW

**Comment by SunPodder** [comment]
Closes #13840

File: packages/gitoxide/build.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+termux_step_make_install() {
+	install -Dm755 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/gix
+	install -Dm755 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/ein
+}
+
+termux_step_install_license(){
+	mkdir -p $TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME
+	cp $TERMUX_PKG_SRCDIR/LICENSE-* $TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME
+}
```
[the remark this answers is not in the document]
Comment (xtkoba) [reply packages/gitoxide/build.sh:34]: (BTW I always do this type of "trivial" modification silently in addition when some other modification is needed in `build.sh`.)

**Comment by xtkoba** [comment]
Why am I an only reviewer? I can only review on the syntactic aspect of this.

The `build.sh` does not have no newline at the end of file. This would cause a superfluous diff output if I modified it afterwards. Not mandatory but it is strongly preferred that you put a newline at EOF.

File: packages/gitoxide/build.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		--target $CARGO_TARGET_NAME \
+		--release \
+		--no-default-features \
+		--features max-pure
+}
+
+termux_step_make_install() {
+	install -Dm755 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/gix
+	install -Dm755 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/ein
+}
```
Comment (xtkoba) [inline packages/gitoxide/build.sh:30]: Put a newline at EOF.

Comment (SunPodder) [reply packages/gitoxide/build.sh]: Done

**Comment by licy183** [comment]
It cannot work properly. Maybe patches on hardcoded paths are needed.

image

**Comment by xtkoba** [comment]
Ah, hardcoded `/etc/resolv.conf`? May require "on-the-fly" patching like 57e2d99ac290f6f239d237a8e9974780f3dc7f83 in `lychee` package.

**Comment by SunPodder** [comment]
Though there are some hardcoded paths, I didn't find any code containing `/etc/resolv.conf`. What's there is `/etc/man.conf` and `/etc/gitconfig`.

```bash
$ grep -rn "/etc/"
Makefile:326:   ./etc/check-package-size.sh
git-config/src/source.rs:71:                .then(|| PathBuf::from(env_var("GIT_CONFIG_SYSTEM").unwrap_or_else(|| "/etc/gitconfig".into())).into()),
git-config/src/source.rs:139:                "file:C:/git-sdk-64/etc/gitconfig  core.symlinks=false\r\nfile:C:/git-sdk-64/etc/gitconfig core.autocrlf=true";
git-config/src/source.rs:140:            let win_cmd = "file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain\r\nfile:C:/Program Files/Git/etc/gitconfigfilter.lfs.clean=git-lfs clean -- %f\r\n";                                              
git-config/src/source.rs:150:                (win_msys, Some("C:/git-sdk-64/etc/gitconfig")),                                                              
git-config/src/source.rs:151:                (win_cmd, Some("C:/Program Files/Git/etc/gitconfig")),
git-config/src/types.rs:21:    /// `$(prefix)/etc/gitconfig` (where prefix is the git-installation directory).
git-config/tests/file/resolve_includes.rs:7:            path = /etc/absolute/missing.config
git-repository/src/repository/permissions.rs:21:    /// This is defined as `$(prefix)/etc/gitconfig` on unix.
git-repository/tests/fixtures/make_config_repo.sh:13:  absolute-path = /etc/man.conf
git-repository/tests/repository/config/identity.rs:120:                Path::new("/etc/man.conf")                                                                               
git-validate/tests/validate/reference.rs:60:        mktest!(any_path_starts_with_slash, b"/etc/foo", RefError::StartsWithSlash);
git-validate/tests/validate/reference.rs:146:        mktest!(any_path_starts_with_slash, b"/etc/foo", RefError::StartsWithSlash);
```

**Comment by xtkoba** [comment]
Thus "on-the-fly". The build system (Cargo) downloads external sources when building. We have to apply patch(es) to that.

**Comment by xtkoba** [comment]
`Cargo.lock` names `trust-dns-resolver` as a dependency. That is what we apply a patch to when building `lychee` package.

**Comment by SunPodder** [comment]
Fixed.

File: packages/gitoxide/trust-dns-resolver.diff [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
++++ b/src/system_conf/unix.rs
+@@ -27,7 +27,7 @@
+ const DEFAULT_PORT: u16 = 53;
+ 
+ pub fn read_system_conf() -> io::Result<(ResolverConfig, ResolverOpts)> {
+-    read_resolv_conf("/etc/resolv.conf")
++    read_resolv_conf("@TERMUX_PREFIX@/etc/resolv.conf")
+ }
+ 
+ fn read_resolv_conf<P: AsRef<Path>>(path: P) -> io::Result<(ResolverConfig, ResolverOpts)> {
```
Comment (xtkoba) [inline packages/gitoxide/trust-dns-resolver.diff:11]: Please put a newline at EOF.

Comment (SunPodder) [reply packages/gitoxide/trust-dns-resolver.diff]: Sorry. I thought I shouldn't manually edit a .diff file 😅

**Comment by licy183** [comment]
`gix clone` works fine.
I'm not very familiar with this tool. It would be great that someone does more functional tests on this package.

**Comment by SunPodder** [comment]
I can't even figure out how to perform add, commit, and push lol

**Comment by xtkoba** [comment]
> `gix clone` works fine.

That will be sufficient. I would test less, just invoking the command without arguments, or `--help` or something.

**Pull request merged**